pynetbox_data_uploader/tests/test_netbox_get_id.py

```diff
@@ -1,39 +1,114 @@
-from unittest.mock import NonCallableMock, patch
-import pytest
-from lib.netbox_api.netbox_get_id import NetboxGetID
-from lib.enums.dcim_device_id import DeviceInfoID
-from lib.enums.dcim_device_no_id import DeviceInfoNoID
-
-
-@pytest.fixture(name="instance")
-def instance_fixture():
-    """
-    This fixture method calls the class being tested.
-    :return: The class object.
-    """
-    netbox = NonCallableMock()
-    return NetboxGetID(netbox)
-
-
-def test_get_id_from_key_with_id_enums(instance):
-    """
-    This test ensures that the get_id method is called for all properties in the DeviceInfoID enum.
-    """
-    with patch("lib.netbox_api.netbox_get_id.NetboxGetID.get_id") as mock_get_id:
-        for member in [prop.name for prop in DeviceInfoID]:
-            mock_dictionary = {member: "abc", "site": "def"}
-            res = instance.get_id_from_key(key=member, dictionary=mock_dictionary)
-            mock_get_id.assert_called()
-            assert res == mock_get_id.return_value
-
-
-def test_get_id_from_key_with_no_id_enums(instance):
-    """
-    This test ensures that the get_id method is not called for all properties in the DeviceInfoNoID enum.
-    """
-    with patch("lib.netbox_api.netbox_get_id.NetboxGetID.get_id") as mock_get_id:
-        for member in [prop.name for prop in DeviceInfoNoID]:
-            mock_dictionary = {member: "abc", "site": "def"}
-            res = instance.get_id_from_key(key=member, dictionary=mock_dictionary)
-            mock_get_id.assert_not_called()
-            assert res == mock_dictionary[member]
+from unittest.mock import patch
+from pytest import fixture
+from lib.netbox_api.netbox_get_id import NetboxGetId
+from lib.utils.device_dataclass import Device
+
+
+@fixture(name="instance")
+def instance_fixture():
+    """
+    This function returns the class to be tested.
+    """
+    mock_api = ""
+    return NetboxGetId(mock_api)
+
+
+device_dict = {
+    "tenant": "t1",
+    "device_role": "dr1",
+    "manufacturer": "m1",
+    "device_type": "dt1",
+    "status": "st1",
+    "site": "si1",
+    "location": "l1",
+    "rack": "r1",
+    "face": "f1",
+    "airflow": "a1",
+    "position": "p1",
+    "name": "n1",
+    "serial": "se1",
+}
```

Comment:
is this a global variable? 
put this in a fixture instead

Reply:
👍